### PR TITLE
test code

### PR DESCRIPTION
**Pull request by vladilyushchenko**

### CODE REVIEW

File: src/test/java/BlockingHashMapTest.java [hunk 1]
```diff
@@ -1,62 +1,64 @@
...
+import java.util.HashMap;
 import java.util.Random;
 
 class BlockingHashMapTest {
-    BlockingHashMap<Integer, Integer> innerMap = new BlockingHashMap<>(10);
-    Random rnd = new Random(System.currentTimeMillis());
-    int[] keys = new int[1000];
-    boolean init = false;
 
     @org.junit.jupiter.api.Test
```
Comment (zaplitny) [inline src/test/java/BlockingHashMapTest.java:8]: import annotation

File: src/main/java/BlockingHashMap.java [hunk 2]
```diff
@@ -57,7 +62,16 @@ public synchronized V get(K key) {
...
+        for (MapEntry<K, V> curr = table[index]; curr != null; curr = curr.next) {
+            if (curr.hash == hash && key.equals(curr.key)) {
+                return true;
+            }
+        }
+
+        return false;
     }
 
     public int hashCode(K x) {
```
Comment (zaplitny) [inline src/main/java/BlockingHashMap.java:77]: private

File: src/main/java/BlockingHashMap.java [hunk 3]
```diff
@@ -57,7 +62,16 @@ public synchronized V get(K key) {
     }
 
     public boolean contains(K key) {
-        return get(key) != null;
+        int hash = hashCode(key);
```
Comment (zaplitny) [inline src/main/java/BlockingHashMap.java:65]: what's wrong with previous implementation? avoid code duplication

File: src/test/java/BlockingHashMapTest.java [hunk 4]
```diff
@@ -1,62 +1,64 @@
...
         thr1.join(); thr2.join();
+
+        for (int key : keys) {
+            innerMap.contains(key);
+        }
+    }
+
+    @org.junit.jupiter.api.Test
+    void testGetReturnsCorrect() {
+        BlockingHashMap<Integer, Integer> innerMap = new BlockingHashMap<>(50);
```
Comment (zaplitny) [inline src/test/java/BlockingHashMapTest.java:40]: why use random? also, I don't see any reason to check so many values. if you want to test some overflow when lot of values inserted - write another test

File: src/test/java/BlockingHashMapTest.java [hunk 5]
```diff
@@ -1,62 +1,64 @@
...
-        }
-        for (int key : keys) {
-            boolean res = innerMap.contains(key);
-            Assertions.assertTrue(res);
-        }
-    }
-
-    void initMaps() throws InterruptedException {
-        init = true;
+    void testPutByTwoThreads() throws InterruptedException {
```
Comment (zaplitny) [inline src/test/java/BlockingHashMapTest.java:9]: what this test is for?

File: src/test/java/BlockingHashMapTest.java [hunk 6]
```diff
@@ -1,62 +1,64 @@
...
         thr1.start(); thr2.start();
         thr1.join(); thr2.join();
+
+        for (int key : keys) {
+            innerMap.contains(key);
+        }
+    }
+
+    @org.junit.jupiter.api.Test
+    void testGetReturnsCorrect() {
```
Comment (zaplitny) [inline src/test/java/BlockingHashMapTest.java:39]: add tests for null and for key that do not exist

**Review by zaplitny — COMMENTED** [review]
- remove .iml file and add it to .gitignore - these files are generated for each user
- no need to pass next in MapEntry's constructor - it's always null in constructor
- add default constructor with predefined size
- add README.md with task description
- add javadocs to class and public methods
- write test that shows why synchronised added
- write test that checks size without adding size method :-)
- update pom.xml - project didn't run when I downloaded it
- add GitHub action that will run test (optional)

File: src/test/java/BlockingHashMapTest.java [hunk 7]
```diff
@@ -1,62 +1,64 @@
...
+            valueByKey.put(keys[i], value);
+        }
+
+        for (int i = 0; i < 1000; i++) {
+            Assertions.assertEquals(innerMap.get(keys[i]), valueByKey.get(keys[i]));
+        }
+    }
+
+    @org.junit.jupiter.api.Test
+    void testPutSameKeys() {
```
Comment (zaplitny) [inline src/test/java/BlockingHashMapTest.java:58]: <Integer, Integer> is bad solution here as key and value are the same. Differentiate keys and values, e.g use strings "key1" / "value1"

File: src/test/java/BlockingHashMapTest.java [hunk 8]
```diff
@@ -1,62 +1,130 @@
...
-import java.util.Random;
+import java.util.*;
 
 class BlockingHashMapTest {
-    BlockingHashMap<Integer, Integer> innerMap = new BlockingHashMap<>(10);
-    Random rnd = new Random(System.currentTimeMillis());
-    int[] keys = new int[1000];
-    boolean init = false;
 
     @org.junit.jupiter.api.Test
```
Comment (zaplitny) [inline src/test/java/BlockingHashMapTest.java:7]: As mentioned import this annotation instead of writing full name with package each time

File: src/test/java/BlockingHashMapTest.java [hunk 9]
```diff
@@ -1,62 +1,130 @@
...
+        BlockingHashMap<Integer, Integer> map = new BlockingHashMap<>(10);
+        map.put(0, 1);
+        map.put(0, 0);
+        Assertions.assertEquals(map.get(0), 0);
     }
 
-    void initMaps() throws InterruptedException {
-        init = true;
+    @org.junit.jupiter.api.Test
+    void testMapHasTrueSize() throws InterruptedException {
```
Comment (zaplitny) [inline src/test/java/BlockingHashMapTest.java:73]: test should be as simple as possible, e.g 

```
BlockinHashMap map = new ...(). 
test(0, map.size();

map.put("key1", "value1");
test(1, map.size();

map.put("key1", "value2");
test(1, map.size();

map.put("key2", "value1");
test(2, map.size();
```

File: src/test/java/BlockingHashMapTest.java [hunk 10]
```diff
@@ -1,62 +1,130 @@
...
+        BlockingHashMap<Integer, Integer> map = new BlockingHashMap<>(10);
+
+        BlockingHashMap<Integer, Integer> innerMap = new BlockingHashMap<>(1);
+        Random rnd = new Random(System.currentTimeMillis());
+        int[] keys = new int[15000];
 
         Thread thr1 = new Thread( () -> {
```
Comment (zaplitny) [inline src/test/java/BlockingHashMapTest.java:80]: Please use thread only if you test threads

File: src/test/java/BlockingHashMapTest.java [hunk 11]
```diff
@@ -1,62 +1,130 @@
 import org.junit.jupiter.api.Assertions;
 
-import java.util.Random;
+import java.util.*;
```
Comment (zaplitny) [inline src/test/java/BlockingHashMapTest.java:3]: Do not use existing List or Map to test functionality
